contracts/solidity/MarketMakers/LMSRMarketMaker.sol

```diff
@@ -13,33 +13,50 @@ contract LMSRMarketMaker is MarketMaker {
      *  Constants
      */
     uint constant ONE = 0x10000000000000000;
+    int constant EXP_LIMIT = 2352680790717288641401;
+
 
     /*
      *  Public functions
      */
-    /// @dev Returns costs to buy given number of outcome tokens
+    /// @dev Returns cost to buy given number of outcome tokens
     /// @param market Market contract
     /// @param outcomeTokenIndex Index of outcome to buy
     /// @param outcomeTokenCount Number of outcome tokens to buy
-    /// @return Returns costs
+    /// @return Returns cost
     function calcCosts(Market market, uint8 outcomeTokenIndex, uint outcomeTokenCount)
         public
         constant
-        returns (uint costs)
+        returns (uint cost)
     {
-        uint[] memory outcomeTokenDistribution = getOutcomeTokenDistribution(market);
-        require(outcomeTokenDistribution.length > 1);
-        uint[2] memory outcomeTokenRange = getOutcomeTokenRange(outcomeTokenDistribution);
-        uint invB = uint(Math.ln(outcomeTokenDistribution.length * ONE)) / 10000;
+        require(market.eventContract().getOutcomeCount() > 0);
+        int[] memory netOutcomeTokensSold = getNetOutcomeTokensSold(market);
+
+        uint logN = uint(Math.ln(netOutcomeTokensSold.length * ONE));
```

Comment:
Needs cast. Throws  `Error: Type int256 is not implicitly convertible to expected type uint256.` otherwise